workaround for PNLib

hkiel opened the pull request:
in PNlib the `DAE.LBINARY` expression `{} and not {}` must be handled and should not be simplified to `false` by `ExpressionSimplify.simplifyLBinary`

hkiel commented:
workaround for #2032

sjoelund commented:
Wouldn't it be better to check for array type in the other cases? You could have crefs which are arrays and here wrong code still

OpenModelica-Hudson commented:
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2017-11-19_08-48-31](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4966/console).

hkiel commented:
I now removed the extra cases again and changed operator check to `DAE.AND(DAE.T_BOOL())` and  `DAE.OR(DAE.T_BOOL())` for the cases where a `DAE.BCONST()` is created.
That should fix the problem in a good manner!?

hkiel commented on Compiler/FrontEnd/ExpressionSimplify.mo:
```diff
@@ -4777,6 +4777,11 @@ algorithm
+    case (_,_,DAE.ARRAY(array={}),_)
+      then origExp;
+    case (_,_,_,DAE.ARRAY(array={}))
+      then origExp;
```
@sjoelund Would it be okay to return the empty arrays in these two cases (instead of `origExp`)?
This would reduce the `{} and not {}` of PNLib to `{}`
I could add these cases again, if that is okay. I don't know if these rules would shadow some error or so.
